hathor/p2p/downloader.py

```diff
@@ -60,6 +60,22 @@ def __init__(self, tx_id: bytes, deferred: Deferred, connections: List['NodeSync
         self.requested_index = 0
         self.retry_count = 0
 
+    def drop_connection(self, connection: 'NodeSyncTimestamp') -> None:
+        """Called by the downloader to remove a connection from the tx_details connections."""
+        if connection not in self.connections:
```

Comment:
Is it better to go through the list twice or just capture the `ValueError` exception on the `self.connections.index(...)` call?

Reply:
Oh, that makes sense. Apparently there is no `list.find` (like there `str.find`) which would be more convenient. But I'll change to handle the exception instead.

Reply:
(although performance-wise, it's not that obvious what is better, exceptions are usually somewhat expensive)